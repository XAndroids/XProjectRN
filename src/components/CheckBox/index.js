import React from 'react'
import {TouchableOpacity, Image} from 'react-native'

import PropTypes from "prop-types";

//函数式组件:XCheckBox
//参考：https://github.com/XLibrarySources/react-native-elements/blob/next/src/checkbox/CheckBox.js
const CheckBox = (props) => {
    const icon = props.isChecked ? props.checkedIcon : props.uncheckedIcon;

    return (
        <TouchableOpacity onPress={props.onCheckedChange}>
            <Image testID={'checkbox_image'} style={[props.style]} source={icon}/>
        </TouchableOpacity>
    );
};

CheckBox.propTypes = {
    isChecked: PropTypes.bool.isRequired,//没有默认值，都声明呈required
    checkedIcon: PropTypes.object.isRequired,//FIXME 这是用PropTypes.object检测是否合适？？
    uncheckedIcon: PropTypes.object.isRequired,
    onCheckedChange: PropTypes.func
};

export default CheckBox
