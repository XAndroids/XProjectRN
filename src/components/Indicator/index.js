import React from 'react'
import {View} from 'react-native'
import PropTypes from 'prop-types'
import {styles} from "./styles"

const Indicator = (props) => {
    let circles = [];
    let {currentIndex, circleCount} = props;

    for (let i = 0; i < circleCount; i++) {
        let style = styles.circle;
        if (i === currentIndex) {
            style = [styles.circle, styles.circleSelected];
        }
        circles.push(<View testID={'indicator_view'} style={style} key={i}/>);
    }

    return (
        <View style={styles.circleContainer}>
            {circles}
        </View>
    )
};

Indicator.propTypes = {
    circleCount: PropTypes.number,
    currentIndex: PropTypes.number,
};

export default Indicator
