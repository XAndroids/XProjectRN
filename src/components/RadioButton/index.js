import React from 'react'
import {TouchableOpacity, View} from 'react-native'
import PropTypes from 'prop-types'
import {styles} from './styles'

const DEFAULT_SIZE_MULTIPLIER = 0.7;
const DEFAULT_OUTER_BORDER_WIDTH_MULTIPLIER = 0.15;
/**
 * 单选按钮
 * 参考：https://github.com/mmazzarolo/react-native-radio-button/blob/master/RadioButton.js
 */
const RadioButton = (props) => {
    const {size, innerColor, outerColor, isSelected, onPress} = props;

    //外圈样式
    const outerStyle = {
        borderColor: outerColor,
        width: size + size * DEFAULT_SIZE_MULTIPLIER,
        height: size + size * DEFAULT_SIZE_MULTIPLIER,
        borderRadius: (size + size * DEFAULT_SIZE_MULTIPLIER) / 2,
        borderWidth: size * DEFAULT_OUTER_BORDER_WIDTH_MULTIPLIER
    };

    //内圈样式
    const innerStyle = {
        width: size,
        height: size,
        borderRadius: size / 2,
        backgroundColor: innerColor
    };

    return (
        <TouchableOpacity testID="radiobutton_touchableopacity_outer" style={[props.style, styles.radio, outerStyle]}
                          onPress={onPress}>
            {isSelected ? <View testID="radiobutton_view_inner" style={innerStyle}/> : null}
        </TouchableOpacity>
    )
};

RadioButton.propTypes = {
    size: PropTypes.number,
    innerColor: PropTypes.string,
    outerColor: PropTypes.string,
    isSelected: PropTypes.bool,
    onPress: PropTypes.func
};

RadioButton.defaultProps = {
    size: 16,
    innerColor: '#058b7d',
    outerColor: '#058b7d',
    isSelected: false,
    onPress: () => null
};

export default RadioButton

