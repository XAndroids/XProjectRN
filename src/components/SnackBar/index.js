import {NativeModules, processColor} from 'react-native';


const SnackBar = {
    //常量
    LENGTH_LONG: NativeModules.RNSnackbar.LENGTH_LONG,
    LENGTH_SHORT: NativeModules.RNSnackbar.LENGTH_SHORT,
    LENGTH_INDEFINITE: NativeModules.RNSnackbar.LENGTH_INDEFINITE,
    /**
     * 显示SnackBar
     */
    show(options) {
        const text = options.text;
        const textColorRaw = options.textColor;
        const textColor = textColorRaw && processColor(textColorRaw);
        const backgroundColor = options.backgroundColor && processColor(options.backgroundColor);

        const action = options.action || {};
        const actionText = action.text;
        const actionTextColorRaw = action.textColor;
        const actionTextColor = actionTextColorRaw && processColor(actionTextColorRaw);

        const onPressCallback = action.onPress || (() => {
        });

        const nativeOptions = {
            ...options,
            text,
            textColor,
            backgroundColor,
            action: options.action ? {
                ...action,
                text: actionText,
                textColor: actionTextColor,
            } : undefined,
        };
        console.log(JSON.stringify(NativeModules));
        NativeModules.RNSnackbar.show(nativeOptions, onPressCallback);
    },

    /**
     * 隐藏SnackBar
     */
    dismiss() {
        NativeModules.RNSnackbar.dismiss();
    },
};

export default SnackBar;
