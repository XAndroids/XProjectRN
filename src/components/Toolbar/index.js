import React from 'react'
import {Image, TextInput, ToolbarAndroid, TouchableOpacity, View} from 'react-native'
import {getWindowsWidth} from '../../utils'
import PropTypes from 'prop-types'

/**
 * Toolbar，Android平台实现
 */
const Toolbar = (props) => {
    return (
        <ToolbarAndroid
            testID={props.testID}
            style={{width: getWindowsWidth(), height: 55, backgroundColor: '#01A3AE'}}
            title={props.title}
            titleColor='white'
            actions={props.actions}
            onActionSelected={props.onActionSelected}>
            {/* 标题栏搜索状态，动态布局搜索框 */}
            {props.isSearch ?
                //FIXME View层级上没有backgroundColor,TextInput的placeholder不显示
                <View style={{flex: 1, flexDirection: 'row', backgroundColor: '#01A3AE'}}>
                    <TouchableOpacity style={{justifyContent: 'center'}}
                                      onPress={props.onPressBack}>
                        <Image source={require('../../../img/ic_action_name.png')}/>
                    </TouchableOpacity>
                    <TextInput style={{flex: 1, marginLeft: 15, fontSize: 18}} placeholder='搜索...'
                               placeholderTextColor='#6cc7cd'/>
                </View> : null}
        </ToolbarAndroid>
    );
};

Toolbar.propTypes = {
    title: PropTypes.string,
    actions: PropTypes.array,
    isSearch: PropTypes.bool,
    onActionSelected: PropTypes.func,
    onPressBack: PropTypes.func
};

export default Toolbar

