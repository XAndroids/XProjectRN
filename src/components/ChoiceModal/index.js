import React, {Component} from 'react'
import {FlatList, Modal, Text, View, TouchableWithoutFeedback} from 'react-native'
import {RadioButton} from '../../components'
import PropTypes from 'prop-types'

/**
 * 单选Modal
 */
export default class ChoiceModal extends Component {
    constructor(props) {
        super(props);
        this._onTypeItemPress = this._onTypeItemPress.bind(this)
    }

    render() {
        const backgroundStyle = {
            backgroundColor: this.props.visible ? 'rgba(0, 0, 0, 0.5)' : '#f5fcff'
        };

        return (
            <Modal testID={this.props.testID}
                   visible={this.props.visible}
                   transparent={true}
                   onRequestClose={this.props.onRequestClose}>
                <View style={[{
                    flex: 1,
                    justifyContent: 'center',
                    paddingLeft: 25,
                    paddingRight: 25,
                    paddingTop: 42,
                    paddingBottom: 42
                }, backgroundStyle]}>
                    <View style={[{backgroundColor: 'white'}]}>
                        <Text style={{
                            height: 45,
                            color: 'black',
                            fontSize: 20,
                            fontWeight: 'bold',
                            textAlignVertical: 'bottom',
                            marginLeft: 30,
                            marginBottom: 8
                        }}>Select Publisher</Text>
                        <FlatList data={this.props.choiceList}
                                  renderItem={this._renderItemComponent}
                                  keyExtractor={(item, index) => index.toString()}/>
                    </View>
                </View>
            </Modal>
        );
    }

    _renderItemComponent = ({item, index}) => {
        return (
            <TouchableWithoutFeedback onPress={() => this._onTypeItemPress(item, index)}>
                <View style={{flexDirection: 'row', height: 47}}>
                    <RadioButton style={{marginLeft: 23, marginRight: 23}} isSelected={this.props.choiceIndex === index}
                                 size={12}/>
                    <Text style={{
                        flex: 1,
                        color: 'black',
                        fontSize: 18,
                        textAlignVertical: 'center'
                    }}>{item.name}</Text>
                </View>
            </TouchableWithoutFeedback>
        );
    };

    _onTypeItemPress(item, index) {
        this.props.onSelectedTypeChanged(item, index);
    }
}

ChoiceModal.defaultProps = {
    visible: false,
    choiceIndex: -1,
};

ChoiceModal.propTypes = {
    testID: PropTypes.string,
    visible: PropTypes.bool,
    choiceList: PropTypes.array.isRequired,
    choiceIndex: PropTypes.number,
    onRequestClose: PropTypes.func,
    onSelectedTypeChanged: PropTypes.func
};
