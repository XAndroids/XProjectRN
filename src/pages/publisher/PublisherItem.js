import React, {Component} from 'react'
import {CardView, CheckBox} from "../../components";
import {Image, Text, View} from "react-native";
import PropTypes from "prop-types";

export default class PublisherItem extends Component {
    constructor(props) {
        super(props);

        this.handleCheckChange = this.handleCheckChange.bind(this);
    }

    render() {
        const {item1} = this.props;

        return (
            <CardView cardElevation={2}
                      backgroundColor={'#00000000'}>
                <View style={{flexDirection: 'row', padding: 6, alignItems: 'center'}}>
                    <Image style={{width: 50, height: 50}}
                           source={require('../../../img/login_icon.png')}/>
                    <View style={{flex: 1, marginLeft: 8}}>
                        <Text style={{fontSize: 18, color: 'black'}}>{item1.name}</Text>
                        <Text
                            style={{
                                fontSize: 12,
                                color: '#73737373'
                            }}>{item1.subscribeNum}</Text>
                    </View>
                    <CheckBox style={{width: 50, height: 50, marginRight: 6}}
                              isChecked={item1.isSubscribed}
                              checkedIcon={require('../../../img/publisherlist_checkbox_selected.png')}
                              uncheckedIcon={require('../../../img/publisherlist_checkbox_add.png')}
                              onCheckedChange={this.handleCheckChange}/>
                </View>
            </CardView>
        );
    }

    handleCheckChange() {
        const {onItemPress, item1} = this.props;
        onItemPress(item1)
    }
}

PublisherItem.propTypes = {
    item1: PropTypes.object.isRequired,
    onItemPress: PropTypes.func
};
