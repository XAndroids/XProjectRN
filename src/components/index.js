import CheckBox from './CheckBox'
import Indicator from './Indicator'
import RadioButton from './RadioButton'
import Toolbar from './Toolbar'
import TypeChoiceModal from './TypeChoiceModal'
import ViewPager from './ViewPager'
import CardView from './CardView'
import SnackBar from "./SnackBar";

export {
    CheckBox,
    Indicator,
    RadioButton,
    Toolbar,
    TypeChoiceModal,
    ViewPager,
    CardView,
    SnackBar
}
