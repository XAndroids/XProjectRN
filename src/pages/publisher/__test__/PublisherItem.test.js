import React from "react";
import {fireEvent, render, shallow} from "react-native-testing-library";
import PublisherItem from "../PublisherItem";
import {CheckBox} from "../../../components";

describe('PublisherItem test', function () {
    const  item1 = {
        publisherId: 'p001',
        type: 't001',
        language: 'l001',
        name: 'The Tech-mock',
        subscribeNum: '970601 subscribers',
        isSubscribed: false
    }

    describe('PublisherItem property test', function () {
        it("item1 property test",function () {
            const {queryByText,queryByType}  = render(<PublisherItem item1={item1}/>)
            const nameText = queryByText(item1.name)
            expect(nameText).not.toBeNull()
            expect(nameText.props.style).toEqual({ fontSize: 18, color: 'black' })
            expect(nameText.props.children).toBe(item1.name)

            const subscribeNumText = queryByText(item1.subscribeNum)
            expect(subscribeNumText).not.toBeNull()
            expect(subscribeNumText.props.style).toEqual({ fontSize: 12, color: '#73737373' })
            expect(subscribeNumText.props.children).toBe(item1.subscribeNum)

            const isSubscribedCheckBox = queryByType(CheckBox)
            expect(isSubscribedCheckBox).not.toBeNull()
            expect(isSubscribedCheckBox.props.isChecked).toBe(false)
            expect(isSubscribedCheckBox.props.style).toEqual( { width: 50, height: 50, marginRight: 6 })
            expect(isSubscribedCheckBox.props.checkedIcon).toEqual({ testUri: '../../../img/publisherlist_checkbox_selected.png' })
            expect(isSubscribedCheckBox.props.uncheckedIcon).toEqual({ testUri: '../../../img/publisherlist_checkbox_add.png' })
        })
    })

    describe('PublisherItem fun test', function () {
        const onItemPress = jest.fn()
        const {queryByType}  = render(<PublisherItem item1={item1} onItemPress={onItemPress}/>)
        const isSubscribedCheckBox = queryByType(CheckBox)
        fireEvent(isSubscribedCheckBox,'onCheckedChange')
        expect(onItemPress).toBeCalledWith(item1)
    })

    describe("PublisherItem Snapshot test",function () {
        const onItemPress = jest.fn()
        const {toJSON}  = render(<PublisherItem item1={item1} onItemPress={onItemPress}/>)
        expect(toJSON()).toMatchSnapshot()
    })

    describe("PublisherItem Shallow test",function () {
        const onItemPress = jest.fn()
        const {output}  = shallow(<PublisherItem item1={item1} onItemPress={onItemPress}/>)
        expect(output).toMatchSnapshot()
    })
})
