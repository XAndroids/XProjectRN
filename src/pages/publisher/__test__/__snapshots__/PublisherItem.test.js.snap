// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 0,
        "shadowColor": "rgba(96,96,96,1)",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.13,
        "shadowRadius": 2,
      },
      Object {},
      Object {
        "backgroundColor": "#00000000",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "padding": 6,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../img/login_icon.png",
        }
      }
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "marginLeft": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 18,
          }
        }
      >
        The Tech-mock
      </Text>
      <Text
        style={
          Object {
            "color": "#73737373",
            "fontSize": 12,
          }
        }
      >
        970601 subscribers
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../img/publisherlist_checkbox_add.png",
          }
        }
        style={
          Array [
            Object {
              "height": 50,
              "marginRight": 6,
              "width": 50,
            },
          ]
        }
        testID="checkbox_image"
      />
    </View>
  </View>
</View>
`;

exports[` 2`] = `
<CardView
  backgroundColor="#00000000"
  cardElevation={2}
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "padding": 6,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../img/login_icon.png",
        }
      }
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
    <Component
      style={
        Object {
          "flex": 1,
          "marginLeft": 8,
        }
      }
    >
      <Component
        style={
          Object {
            "color": "black",
            "fontSize": 18,
          }
        }
      >
        The Tech-mock
      </Component>
      <Component
        style={
          Object {
            "color": "#73737373",
            "fontSize": 12,
          }
        }
      >
        970601 subscribers
      </Component>
    </Component>
    <CheckBox
      checkedIcon={
        Object {
          "testUri": "../../../img/publisherlist_checkbox_selected.png",
        }
      }
      isChecked={false}
      onCheckedChange={[Function]}
      style={
        Object {
          "height": 50,
          "marginRight": 6,
          "width": 50,
        }
      }
      uncheckedIcon={
        Object {
          "testUri": "../../../img/publisherlist_checkbox_add.png",
        }
      }
    />
  </Component>
</CardView>
`;
